packages/time-picker/test/value-commit.test.js

```diff
@@ -0,0 +1,391 @@
+      await sendKeys({ press: 'Escape' });
+      expectNoValueCommit();
+    });
+
+    it('should not commit but validate on close with outside click', () => {
+      timePicker.click();
+      outsideClick();
+      expectValidationOnly();
+    });
+
+    it('should not commit but validate on close with Escape', async () => {
+      timePicker.click();
+      await sendKeys({ press: 'Escape' });
+      expectValidationOnly();
+    });
+
+    it('should not commit on ArrowDown', async () => {
+      // Open the dropdown
+      await sendKeys({ press: 'ArrowDown' });
+      // Focus an item
+      await sendKeys({ press: 'ArrowDown' });
+      expectNoValueCommit();
+    });
+
+    it('should not commit on ArrowUp', async () => {
+      // Open the dropdown
+      await sendKeys({ press: 'ArrowUp' });
+      // Focus an item
+      await sendKeys({ press: 'ArrowUp' });
+      expectNoValueCommit();
+    });
+  });
+
+  describe('entering parsable input', () => {
+    beforeEach(async () => {
+      await sendKeys({ type: '12:00' });
+    });
+
+    it('should not commit by default', () => {
+      expectNoValueCommit();
+    });
+
+    it('should commit on blur', () => {
+      timePicker.blur();
+      expectValueCommit('12:00');
+    });
+
+    it('should commit on Enter', async () => {
+      await sendKeys({ press: 'Enter' });
+      expectValueCommit('12:00');
+    });
+
+    it('should commit on close with outside click', () => {
+      outsideClick();
+      expectValueCommit('12:00');
+    });
+
+    it('should revert and validate on close with Escape', async () => {
+      await sendKeys({ press: 'Escape' });
+      await sendKeys({ press: 'Escape' });
+      expectValidationOnly();
+      expect(timePicker.inputElement.value).to.equal('');
+    });
+  });
+
+  describe('entering unparsable input', () => {
+    beforeEach(async () => {
+      await sendKeys({ type: 'INVALID' });
+    });
+
+    it('should not commit by default', () => {
+      expectNoValueCommit();
+    });
+
+    it('should not commit but validate on blur', () => {
+      timePicker.blur();
+      expectValidationOnly();
+      expect(timePicker.inputElement.value).to.equal('INVALID');
+    });
+
+    it('should not commit but validate on Enter', async () => {
+      await sendKeys({ press: 'Enter' });
+      expectValidationOnly();
+      expect(timePicker.inputElement.value).to.equal('INVALID');
+    });
+
+    it('should not commit but validate on close with outside click', () => {
+      outsideClick();
+      expectValidationOnly();
+      expect(timePicker.inputElement.value).to.equal('INVALID');
+    });
+
+    it('should revert and validate on close with Escape', async () => {
+      await sendKeys({ press: 'Escape' });
+      await sendKeys({ press: 'Escape' });
+      expectValidationOnly();
+      expect(timePicker.inputElement.value).to.equal('');
+    });
+
+    describe('clearing input with Backspace', () => {
```

Comment:
Should we consider `clearing input with Backspace` for parseable input tests as well?

Reply:
Actually, there is already `with value -> clearing input with Backspace` that covers that. However, would you prefer it to be placed inside `entering parsable input`, same as `entering unparsable input`?

Reply:
Another option would be to flatten the structure:

- entering parsable input
- entering unparsable input
- with parsable input committed
- with unparsable input committed (doesn't sound very nice considering that there is currently no value commit in this case)
- with value set programmatically

Reply:
Done.

The structure now is the following:

```
> parsable input entered
> parsable input committed
  > input cleared with Backspace
> unparsable input entered
> unparsable input commited
  > input cleared with Backspace
> value set programmatically
  > parsable input entered
  > unparsable input entered
> with clear button
> with step
```